torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -393,7 +398,7 @@ void DistEngine::cleanupBackwardPass(const ContextPtr& autogradContext) {
   // not leaking any references to the gradients anywhere else.
   const auto& futureGrads =
       autogradContext->retrieveGraphTask()->future_result_;
-  TORCH_INTERNAL_ASSERT(futureGrads.use_count() == 1);
+  TORCH_INTERNAL_ASSERT(futureGrads.use_count() <= 2);
```

Comment:
This needs to be == 1 and was added so that we don't leak references to the gradients. What is the reason this needs to be changed to <=2?

Reply:
The 2 is because of the second future reference (since the weak arg becomes strong).
Do you have a preference for a better alternative?

Reply:
Actually, you might prefer what I send in a few minutes